Dockerfile

```diff
@@ -0,0 +1,28 @@
+FROM python:3.8-slim-buster AS base
+LABEL key="Jeremy Brex todo_app"
+
+# Add Curl as not part of slim Buster
+RUN apt-get update -y && apt-get install curl -y
+
+# Install Poetry
+
+ENV POETRY_HOME=/poetry
+ENV PATH=${POETRY_HOME}/bin:${PATH}
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python
+
+# Set the working directory.
+WORKDIR /app
+COPY todo_app /app/todo_app
+COPY pyproject.toml /app
+COPY poetry.toml /app
+
+RUN poetry install
```

Comment:
I would suggest moving this into the individual stages.

So dev can do `poetry install` while production can do something like `RUN poetry config virtualenvs.create false --local && poetry install --no-dev`
This means production will skip creating a virtual environment, and will skip installing dev dependencies like pytest. The virtual environment is relevant to module 8 (running in it Heroku) because of some weirdness with how Heroku runs containers. 

But note, currently you incorrectly have "requests" in the dev dependencies section.

Reply:
have done but need to review this comment in more detail with you